### DISCUSSION TITLE
How to access S3 with a temporary credential

### DISCUSSION THREAD

**Discussion by joaocasarin** [soto-project/soto#681] [Q&A]
Well, I need to access the objects in a S3 bucket using our Vapor API, but the issue is that our aws_secret_key and aws_secret_key lasts only for 8 hours. Is there any other alternative like using a bucket policy?

**Comment by adam-fowler** [comment]
Not totally sure what you are trying to do here. I have a few of questions. 
- Where are you getting your credentials from? 
- Is there a method to renew them?
- Are you running your Vapor server on AWS?

**Reply by joaocasarin** [reply]
Hi Adam, sorry the delay and lack of informations!

Actually I am really new to Swift and AWS as well and didn't know exactly what I needed to share.

basically we can't use static keys, the only way would be using custom IAM role in AWS and grant that role the permission to assume a STS role with access to our bucket. Not sure how to do this with `soto`, but I ended up using typescript which also failed, and then Python which succeeded!

But thank you anyways!!! All success for you and the project!

**Reply by adam-fowler** [reply]
AWS has so many ways to generate credentials it can be confusing.

**Reply by joaocasarin** [reply]
yea, that's absolutely true and not being able to use our keys because they expire so quickly makes it really harder to setup the correct credentials in our project...
